hack/latest-binaries.sh

```diff
@@ -13,7 +13,7 @@ MINOR_VERSION="${1}"
 
 # retrieve the available "VERSION/BUILD_DATE" prefixes (e.g. "1.28.1/2023-09-14")
 # from the binary object keys, sorted in descending semver order, and pick the first one
-LATEST_BINARIES=$(aws s3api list-objects-v2 --bucket amazon-eks --prefix "${MINOR_VERSION}" --query 'Contents[*].[Key]' --output text | cut -d'/' -f-2 | sort -Vru | head -n1)
+LATEST_BINARIES=$(aws s3api list-objects-v2 --bucket amazon-eks --prefix "${MINOR_VERSION}" --query 'Contents[*].[Key]' --output text --region us-west-2 --no-sign-request | cut -d'/' -f-2 | sort -Vru | head -n1)
```

Comment:
Let's add a big `TODO` outlining the long term solution (being able to use `--region $binary_bucket_region`)

Reply:
This is also forcing customers in `aws-cn` to query the bucket in `aws`, which isn't great.

Reply:
per https://github.com/awslabs/amazon-eks-ami/pull/1641/files#r1486814106

whilst this is not "great" it is also not bad.  This is just retrieving metadata - and even if that takes an extra 5 seconds, it disappears into the time taken to actually build the image.